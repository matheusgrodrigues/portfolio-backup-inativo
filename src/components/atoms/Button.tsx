import styled from 'styled-components'

interface ButtonProps {
    $background: 'color_white' | 'color_gradient_primary600'
    $fontWeight: 'fw_semibold'
    $fontSize: 'fs_textMd'
    $variant: 'lg'
    $color: 'color_white' | 'color_gray500'
}

export const Button = styled.button<ButtonProps>`
    border-radius: 8px;
    border: none;
    cursor: pointer;
    width: 100%;

    ${(props) => {
        if (props.$variant === 'lg') {
            return `
                height: 44px;
                width: 172px;
            `
        }
    }}

    font-weight: ${(props) => props.theme.ref.font[props.$fontWeight]};
    background: ${(props) => props.theme.ref.colors[props.$background]};
    font-size: ${(props) => props.theme.ref.font[props.$fontSize]};
    color: ${(props) => props.theme.ref.colors[props.$color]};
`

export const ButtonTransparent = styled.button`
    background: transparent;
    border: none;
    cursor: pointer;
`
