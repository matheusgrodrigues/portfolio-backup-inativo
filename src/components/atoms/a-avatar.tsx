import { styled } from "styled-components";

import * as RadixAvatar from "@radix-ui/react-avatar";

interface IAvatarRoot extends RadixAvatar.AvatarProps {
  "data-testid"?: string;
}

const AvatarRoot = styled(RadixAvatar.Root).attrs<IAvatarRoot>(
  (props) => ({
    as: "span",
    "data-testid": props["data-testid"] || "a-avatar",
  })
)`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  vertical-align: middle;
  overflow: hidden;
  user-select: none;
  width: 45px;
  height: 45px;
  border-radius: 100%;
  background-color: var(--black-a3);
`;

interface IAvatarImage extends RadixAvatar.AvatarImageProps {
  "data-testid"?: string;
}

const AvatarImage = styled(RadixAvatar.Image).attrs<IAvatarImage>(
  (props) => ({
    as: "img",
    "data-testid": props["data-testid"] || "a-avatar-image",
  })
)`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: inherit;
`;

interface IAvatarFallBack extends RadixAvatar.AvatarFallbackProps {
  "data-testid"?: string;
}

const AvatarFallback = styled(
  RadixAvatar.Fallback
).attrs<IAvatarFallBack>((props) => ({
  as: "span",
  "data-testid": props["data-testid"] || "a-avatar-fallback",
}))`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: white;
  color: var(--violet-11);
  font-size: 15px;
  line-height: 1;
  font-weight: 500;
`;

interface IAvatar extends RadixAvatar.AvatarProps {
  src: string;
  alt: string;
}

export function Avatar({ src, alt, ...props }: IAvatar) {
  return (
    <AvatarRoot {...props}>
      <AvatarImage src={src} alt={alt} />
    </AvatarRoot>
  );
}
