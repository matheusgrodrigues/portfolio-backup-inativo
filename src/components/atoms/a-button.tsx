import styled from "styled-components";

interface IAButton {
  $variant: "lg";
  $background: "color_white" | "color_gradient_primary600";
  $fontSize: "fs_textMd";
  $fontWeight: "fw_semibold";
  $color: "color_white" | "color_gray500";
}

export const AButton = styled.button<IAButton>`
  width: 100%;
  border-radius: 8px;
  border: none;
  cursor: pointer;

  ${(props) => {
    switch (props.$variant) {
      case "lg":
        return `
            max-width: 172px;
            height: 44px;
            `;
        break;
    }
  }}

  background: ${(props) => props.theme.ref.colors[props.$background]};
  color: ${(props) => props.theme.ref.colors[props.$color]};
  font-weight: ${(props) => props.theme.ref.font[props.$fontWeight]};
  font-size: ${(props) => props.theme.ref.font[props.$fontSize]};
`;
