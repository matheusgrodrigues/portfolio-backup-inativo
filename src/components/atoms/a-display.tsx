import { ReactNode } from "react";
import styled from "styled-components";

// Theme
import { themeLight } from "@/src/theme";
import { ThemeProvider } from "@/src/theme/helpers/theme-provider";

interface IADisplay {
  $variant: "lg";
  fontWeight: "semibold";
  color: "gradient" | "gray900";
  children: ReactNode;
}

const Display = styled("h1").attrs<IADisplay>((props) => ({}))`
  ${(props) => {
    const { fs_displayLg } = props.theme.ref.font;

    switch (props.$variant) {
      case "lg":
        return `
        font-size: ${fs_displayLg};
        letter-spacing: -0.96px;
      `;
        break;
    }
  }};

  ${(props) => {
    const { color_primary600, color_gray900, color_gradient_primary600 } =
      props.theme.ref.colors;

    switch (props.color) {
      case "gradient":
        return `
     color: ${color_gradient_primary600}
   `;
        break;
      case "gray900":
        return `
       color: ${color_gray900}
     `;
        break;
    }
  }};
`;

export const ADisplay = ({
  $variant,
  fontWeight,
  color,
  children,
}: IADisplay) => {
  return (
    <>
      {$variant === "lg" && (
        <ThemeProvider theme={themeLight}>
          <Display
            data-testid="a-display"
            $variant={$variant}
            fontWeight={fontWeight}
            color={color}
          >
            {children}
          </Display>
        </ThemeProvider>
      )}
    </>
  );
};
