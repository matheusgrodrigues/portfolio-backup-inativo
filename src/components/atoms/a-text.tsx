import { ReactNode } from "react";
import styled from "styled-components";

// Theme
import { themeLight } from "@/src/theme";
import { ThemeProvider } from "@/src/theme/helpers/theme-provider";

interface IAText {
  $variant: "sm" | "md" | "lg" | "xl";
  fontWeight: "regular" | "semibold";
  color: "white" | "black" | "gradient" | "gray50" | "gray500" | "gray900";
  children: ReactNode;
}

const ATextStyled = styled("p").attrs<IAText>((props) => ({}))`
  ${(props) => {
    const { fw_regular, fw_semibold } = props.theme.ref.font;

    switch (props.fontWeight) {
      case "regular":
        return `font-weight: ${fw_regular}`;
        break;
      case "semibold":
        return `font-weight: ${fw_semibold}`;
        break;
      default:
        return `font-weight: initial`;
        break;
    }
  }}

  ${(props) => {
    const { fs_textSm, fs_textMd, fs_displayLg, fs_textXl } =
      props.theme.ref.font;

    switch (props.$variant) {
      case "lg":
        return `font-size: ${fs_displayLg};
        letter-spacing: -0.96px;`;
        break;
      case "xl":
        return `font-size: ${fs_textXl}`;
        break;
      case "md":
        return `font-size: ${fs_textMd}`;
        break;
      case "sm":
        return `font-size: ${fs_textSm}`;
        break;
    }
  }};

  ${(props) => {
    const {
      color_primary50,
      color_primary500,
      color_primary600,
      color_primary900,
      white,
      black,
    } = props.theme.ref.colors;

    switch (props.color) {
      case "gradient":
        return `color: ${color_primary600}`;
        break;
      case "gray50":
        return `color: ${color_primary50}`;
      case "gray500":
        return `color: ${color_primary500}`;
        break;
      case "gray900":
        return `color: ${color_primary900}`;
        break;
      case "black":
        return `color: ${white}`;
      default:
        return `color: ${black}`;
    }
  }};
`;

export const AText = ({ $variant, fontWeight, color, children }: IAText) => {
  return (
    <ThemeProvider theme={themeLight}>
      <ATextStyled
        data-testid="a-text"
        $variant={$variant}
        fontWeight={fontWeight}
        color={color}
      >
        {children}
      </ATextStyled>
    </ThemeProvider>
  );
};
