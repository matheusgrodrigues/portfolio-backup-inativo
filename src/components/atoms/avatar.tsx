import { styled } from "styled-components";

import * as RadixAvatar from "@radix-ui/react-avatar";

interface IAvatarRoot extends RadixAvatar.AvatarProps {
  "data-testid"?: string;
}

const AvatarRoot = styled(RadixAvatar.Root).attrs<IAvatarRoot>((props) => ({
  as: "span",
  "data-testid": props["data-testid"],
}))`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  vertical-align: middle;
  overflow: hidden;
  user-select: none;
  width: 45px;
  height: 45px;
  border-radius: 100%;
  background-color: ${({ theme }) => theme.ref.colors.color_gray500};
`;

interface IAvatarImage extends RadixAvatar.AvatarImageProps {
  "data-testid"?: string;
}

const AvatarImage = styled(RadixAvatar.Image).attrs<IAvatarImage>((props) => ({
  as: "img",
  "data-testid": props["data-testid"],
}))`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: inherit;
`;

interface IAvatar extends RadixAvatar.AvatarProps {
  src: string;
  alt: string;
}

export function Avatar({ src, alt, ...props }: IAvatar) {
  return (
    <AvatarRoot data-testid="a-avatar" {...props}>
      <AvatarImage data-testid="a-avatar-image" src={src} alt={alt} />
    </AvatarRoot>
  );
}
