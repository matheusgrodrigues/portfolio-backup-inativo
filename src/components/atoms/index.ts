import { AAvatar } from "./a-avatar/a-avatar";

export { AAvatar };
