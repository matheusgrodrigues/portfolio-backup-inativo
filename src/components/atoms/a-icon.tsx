import {
  SunIcon,
  LinkedInLogoIcon,
  GitHubLogoIcon,
} from "@radix-ui/react-icons";

import { IconProps } from "@radix-ui/react-icons/dist/types";

// Atoms
import { AButtonTransparent } from "./a-button";

type TAIcon = "sun" | "linkedin-rounded" | "github-rounded" | "github-square";

interface IAIcon extends IconProps {
  icon: TAIcon;
  callback?: () => void;
}

export const AIcon = ({ icon, callback, ...props }: IAIcon) => {
  return (
    <AButtonTransparent onClick={callback}>
      {icon === "sun" && <SunIcon width={"200px"} {...props} />}
      {icon === "linkedin-rounded" && <LinkedInLogoIcon {...props} />}
      {icon === "github-rounded" && <LinkedInLogoIcon {...props} />}
      {icon === "github-square" && <GitHubLogoIcon {...props} />}
    </AButtonTransparent>
  );
};
