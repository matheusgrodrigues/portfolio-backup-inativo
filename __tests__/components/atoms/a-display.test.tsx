import renderer from "react-test-renderer";

// Atoms
import { ADisplay } from "@/src/components/atoms/a-display";

describe("Deve renderizar o ADisplay corretamente", () => {
  it("Deve preservar a estrutura visual do ADisplay", () => {
    const three = renderer
      .create(
        <ADisplay variant="lg" fontWeight="semibold" color="gradient">
          ADisplay
        </ADisplay>
      )
      .toJSON();

    expect(three).toMatchSnapshot();
  });
});
