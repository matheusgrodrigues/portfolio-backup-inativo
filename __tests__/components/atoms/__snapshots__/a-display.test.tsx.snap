// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o ADisplay corretamente Deve preservar a estrutura visual do ADisplay 1`] = `null`;
